karbon/karbon_api.go

```diff
@@ -22,6 +24,69 @@ type Client struct {
 	Meta            MetaService
 }
 
+type ClientService interface {
+	CreateK8sRegistration(createRequest *K8sCreateClusterRegistrationRequest) (*K8sCreateClusterRegistrationResponse, error)
+	DeleteK8sRegistration() (*K8sClusterRegistrationDeleteResponse, error)
+	GetK8sRegistration() (*K8sClusterRegistration, error)
+	GetK8sRegistrationList() (*K8sClusterRegistrationList, error)
+}
+
+func (op Client) CreateK8sRegistration(createRequest *K8sCreateClusterRegistrationRequest) (*K8sCreateClusterRegistrationResponse, error) {
+	ctx := context.TODO()
```

Comment:
let's not use a `context.TODO`; Context should be part of the method parameters and should be transparently passed further on.